### PR TITLE
give action performer environment variables

### PR DESCRIPTION
**Pull request by jeberl**
Summary
---------

This is needed to read actioner configs from the configs table

Test Plan
---------

Pushed to aws, ran with the following event, and saw that no errors emerged
```
{
  "Records": [
    {
      "body": "{\"ContentKey\": \"key\", \"ContentHash\": \"hash\", \"MatchingBankedSignals\": [{\"BankedContentId\": \"2862392437204724\", \"BankId\": \"bank 4\", \"BankSource\": \"te\", \"Classifications\": []}, {\"BankedContentId\": \"4194946153908639\", \"BankId\": \"bank 4\", \"BankSource\": \"te\", \"Classifications\": []}], \"ActionLabelValue\": \"EnqueForReview\"}"}
  ]
}
```

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 1]
```diff
@@ -29,12 +54,41 @@ def lambda_handler(event, context):
...
+        ),
+    ]
+    for performer_config in configs:
+        config.update_config(performer_config)
+
+    action_message = ActionMessage(
+        "key",
+        "hash",
+        matching_banked_signals=banked_signals,
+        action_label=ActionLabel("EnqueForReview"),
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:91]: Enqueue? (or are we saving chars?

File: hasher-matcher-actioner/terraform/actions/.terraform.lock.hcl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# This file is maintained automatically by "terraform init".
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/actions/.terraform.lock.hcl:1]: what does this have a sub dir lock? it feels like this is an error but I'm not sure

**Review by BarrettOlson — APPROVED** [review]
Yay testability! (we should maybe make note of all the __main__ lambdas we want to refactor) 
Accepting to unblock 
Added a nit in line, please remove the extra lockfile unless you are sure it is needed.

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 3]
```diff
@@ -29,12 +54,41 @@ def lambda_handler(event, context):
     an action message on the actions queue and here's where they're popped
     off and dealt with.
     """
+    lambda_init_once()
     for sqs_record in event["Records"]:
         # TODO research max # sqs records / lambda_handler invocation
-        action_message = ActionMessage.from_aws_message(json.loads(sqs_record["body"]))
+        action_message = ActionMessage.from_aws_message(sqs_record["body"])
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:60]: not json anymore I take it?

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 4]
```diff
@@ -29,12 +54,41 @@ def lambda_handler(event, context):
...
 
     return {"action_performed": "true"}
+
+
+if __name__ == "__main__":
+    lambda_init_once()
+
+    banked_signals = [
+        BankedSignal("2862392437204724", "bank 4", "te"),
+        BankedSignal("4194946153908639", "bank 4", "te"),
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:74]: should the bank names be different?

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 5]
```diff
@@ -1,19 +1,44 @@
...
+    Lambda initialization is weird - despite the existence of perfectly
+    good constructions like __name__ == __main__, there don't appear
+    to be easy ways to split your lambda-specific logic from your
+    module logic except by splitting up the files and making your
+    lambda entry as small as possible.
+
+    TODO: Just refactor this file to separate the lambda and functional
+          components
+    """
+    collab_config_table = os.environ["HMA_CONFIG_TABLE"]
```
Comment (eriksendc) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:38]: collab_config_table --> config_table_name
HMA_CONFIG_TABLE --> CONFIG_TABLE_NAME

File: hasher-matcher-actioner/terraform/actions/main.tf [hunk 6]
```diff
@@ -96,8 +96,9 @@ resource "aws_lambda_function" "action_evaluator" {
 
   environment {
     variables = {
-      ACTIONS_QUEUE_URL = aws_sqs_queue.actions_queue.id,
+      ACTIONS_QUEUE_URL   = aws_sqs_queue.actions_queue.id,
       REACTIONS_QUEUE_URL = aws_sqs_queue.reactions_queue.id,
+      HMA_CONFIG_TABLE    = var.hma_config_table.name
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/actions/main.tf:101]: watch for my code coming in with this change but named a different way. here and further on.

File: hasher-matcher-actioner/terraform/actions/main.tf [hunk 7]
```diff
@@ -116,6 +117,12 @@ resource "aws_lambda_function" "action_performer" {
 
   timeout     = 300
   memory_size = 512
+
+  environment {
+    variables = {
+      HMA_CONFIG_TABLE = var.hma_config_table.name
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/actions/main.tf:123]: This won't come in with my changes but you should use the naming from above.

File: hasher-matcher-actioner/terraform/actions/main.tf [hunk 8]
```diff
@@ -265,6 +272,14 @@ data "aws_iam_policy_document" "action_performer" {
     actions   = ["secretsmanager:GetSecretValue"]
     resources = [var.te_api_token_secret.arn]
   }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "dynamodb:Get*",
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/actions/main.tf:279]: The permission I used was scan, not get. Got that from the fetcher I though?

**Review by eriksendc — CHANGES_REQUESTED** [review]
I'd like to re-review after you rebase / merge changes in from master. Besides resolving conflicts, the only blocking issue is regarding the permissions you set up for your lambda to read from the config table.

File: hasher-matcher-actioner/terraform/actions/.terraform.lock.hcl (hunk 2 above)
Comment (jeberl) [reply hasher-matcher-actioner/terraform/actions/.terraform.lock.hcl:1]: deleted, ran terraform init, and terraform apply. Everything seems fine

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py (hunk 3 above)
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:60]: The `from_aws_message` includes json.loads

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py (hunk 4 above)
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:74]: Never read? I think two matches in the same bank is possible

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py (hunk 5 above)
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py]: Changed

File: hasher-matcher-actioner/terraform/actions/main.tf (hunk 8 above)
Comment (jeberl) [reply hasher-matcher-actioner/terraform/actions/main.tf:284]: Hmm. Get* works for me. Any reason to change?

Comment (eriksendc) [reply hasher-matcher-actioner/terraform/actions/main.tf:284]: consistency?

**Review by eriksendc — APPROVED** [review]

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py (hunk 4 above)
Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:74]: bank == dataset? or is bank a new concept?

**Pull request merged**